File: ios/Harakat Lens/Config/ArabicConfig.swift
//
//  ArabicConfig.swift
//  Harakat Lens
//
//  The complete `BiangBiangConfig` for Harakat Lens: one Arabic
//  `LanguageProfile` (single ICU-transliterator variant) plus the
//  `QuranPlugin` and a "Quran mode" extra setting. The library renders
//  every screen from this data.
//

import BiangBiangUI

enum ArabicConfig {
    @MainActor
    static let arabicConfig: BiangBiangConfig = .init(
        branding: Branding(
            appName: "Harakat Lens",
            accentColorHex: "#006C35",
            logoAssetName: "Logo",
            buttonLogoAssetName: "Logo",
            githubRepo: "veeso/Harakat-Lens",
            supportEmail: "info@veeso.dev",
            appStoreId: "6767614189",
            playStoreId: "dev.veeso.harakatlens"
        ),
        languages: [
            LanguageProfile(
                id: "arabic",
                displayName: "Arabic",
                scriptRanges: [0x0600 ... 0x06FF],
                ocrRecognizer: .arabic,
                variants: [
                    LanguageVariant(
                        id: "arabic",
                        displayName: "Arabic",
                        transliterator: ArabicTransliterator(),
                        ttsLanguageCode: "ar",
                        translatable: true
                    ),
                ]
            ),
        ],
        extraSettings: [
            SettingDescriptor(
                key: "quranMode",
                kind: .toggle,
                label: "Quran mode",
                defaultValue: "false"
            ),
        ],
        plugins: [QuranPlugin()],
        features: FeatureFlags(),
        strings: [
            "inputTitle": "Arabic",
            "outputTitle": "Transliteration",
            "appSubtitle": "Transliterate Arabic",
        ]
    )
}

File: android/app/src/main/java/dev/veeso/harakatlens/arabic/ArabicConfig.kt
package dev.veeso.harakatlens.arabic

import android.content.Context
import dev.veeso.biangbiangui.config.BiangBiangConfig
import dev.veeso.biangbiangui.config.Branding
import dev.veeso.biangbiangui.config.FeatureFlags
import dev.veeso.biangbiangui.config.LanguageProfile
import dev.veeso.biangbiangui.config.LanguageVariant
import dev.veeso.biangbiangui.config.OcrRecognizer
import dev.veeso.biangbiangui.config.SettingDescriptor
import dev.veeso.harakatlens.arabic.quran.QuranPlugin

/**
 * A complete [BiangBiangConfig] proving the library's plugin slot: one Arabic
 * [LanguageProfile] (single ICU-transliterator variant) plus the
 * [QuranPlugin] and a "Quran mode" extra setting. Branding mirrors the
 * Harakat-Lens app.
 *
 * Mirrors iOS `ArabicConfig.arabicConfig` exactly (same branding,
 * scriptRanges, ocrRecognizer, single `ar`/translatable variant, `quranMode`
 * toggle descriptor, single [QuranPlugin]). The iOS config is a static `let`;
 * on Android it is a factory function because the transliterator / plugin
 * need a [Context] for asset loading (`Bundle.module` -> `Context.assets`).
 */
fun arabicConfig(context: Context): BiangBiangConfig {
    val appContext = context.applicationContext
    val transliterator = ArabicTransliterator(
        Vocalizer(VocalizationDictionary.get(appContext)),
    )
    return BiangBiangConfig(
        branding = Branding(
            appName = "Harakat Lens",
            accentColorHex = "#006C35",
            logoAssetName = "logo",
            buttonLogoAssetName = "logo_button_ico",
            githubRepo = "veeso/Harakat-Lens",
            supportEmail = "info@veeso.dev",
            appStoreId = "6767614189",
            playStoreId = "dev.veeso.harakatlens",
        ),
        languages = listOf(
            LanguageProfile(
                id = "arabic",
                displayName = "Arabic",
                scriptRanges = listOf(0x0600u..0x06FFu),
                ocrRecognizer = OcrRecognizer.ARABIC,
                variants = listOf(
                    LanguageVariant(
                        id = "arabic",
                        displayName = "Arabic",
                        transliterator = transliterator,
                        ttsLanguageCode = "ar",
                        translatable = true,
                    ),
                ),
                ocrService = TesseractOcrService(appContext),
            ),
        ),
        minimumOcrScaleFactor = 0.5f,
        extraSettings = listOf(
            SettingDescriptor(
                key = "quranMode",
                kind = SettingDescriptor.Kind.Toggle,
                label = "Quran mode",
                defaultValue = "false",
            ),
        ),
        plugins = listOf(QuranPlugin(appContext)),
        features = FeatureFlags(),
        strings = mapOf(
            "inputTitle" to "Arabic",
            "outputTitle" to "Transliteration",
            "appSubtitle" to "Transliterate Arabic",
        ),
    )
}
